php/Blockonomics.php

```diff
@@ -79,23 +92,119 @@ public function get_price($currency, $crypto)
+    public function update_callback($callback_url, $crypto, $xpub)
     {
-        $url = Blockonomics::SET_CALLBACK_URL;
+        if ($crypto == 'btc'){
+            $url = Blockonomics::SET_CALLBACK_URL;
+        }else{
+            $url = Blockonomics::BCH_SET_CALLBACK_URL;
+        }
         $body = json_encode(array('callback' => $callback_url, 'xpub' => $xpub));
         $response = $this->post($url, $this->api_key, $body);
         return json_decode(wp_remote_retrieve_body($response));
     }
 
-    public function get_callbacks()
+    public function get_callbacks($crypto)
     {
-        $url = Blockonomics::GET_CALLBACKS_URL;
+        if ($crypto == 'btc'){
+            $url = Blockonomics::GET_CALLBACKS_URL;
+        }else{
+            $url = Blockonomics::BCH_GET_CALLBACKS_URL;
+        }
         $response = $this->get($url, $this->api_key);
         return $response;
     }
+    
+    public function check_get_callbacks_response_code($response){
+        $error_str = '';
+        //TODO: Check This: WE should actually check code for timeout
+        if (!wp_remote_retrieve_response_code($response)) {
+            $error_str = __('Your server is blocking outgoing HTTPS calls', 'blockonomics-bitcoin-payments');
+        }
+        elseif (wp_remote_retrieve_response_code($response)==401)
+            $error_str = __('API Key is incorrect', 'blockonomics-bitcoin-payments');
+        elseif (wp_remote_retrieve_response_code($response)!=200)
+            $error_str = $response->data;
+        return $error_str;
+    }
+
+    public function check_get_callbacks_response_body ($response, $crypto){
+        $error_str = '';
+        $response_body = json_decode(wp_remote_retrieve_body($response));
+
+        $callback_secret = get_option('blockonomics_callback_secret');
+        $api_url = WC()->api_request_url('WC_Gateway_Blockonomics');
+        $callback_url = add_query_arg('secret', $callback_secret, $api_url);
+        $callback_url_without_schema = preg_replace('/https?:\/\//', '', $callback_url);
+
+        if (!isset($response_body) || count($response_body) == 0)
+        {
+            $error_str = __('You have not entered an xPub', 'blockonomics-bitcoin-payments');
+        }
+        elseif (count($response_body) == 1)
+        {
+            $response_callback = '';
+            $response_address = '';
+            if(isset($response_body[0])){
+                $response_callback = isset($response_body[0]->callback) ? $response_body[0]->callback : '';
+                $response_address = isset($response_body[0]->address) ? $response_body[0]->address : '';
+            }
+            $response_callback_without_schema = preg_replace('/https?:\/\//', '', $response_callback);
+            if(!$response_callback || $response_callback == null)
+            {
+                //No callback URL set, set one 
+                $this->update_callback($callback_url, $response_address, $crypto);
```

Comment:
update_callback should receive `$callback_url, $crypto, $xpub`, however we are passing `$callback_url, $response_address, $crypto`. This leads to "Could not find matching xpub." error.

Reply:
Fixed